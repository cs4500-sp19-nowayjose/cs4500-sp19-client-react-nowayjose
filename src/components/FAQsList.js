import React from 'react'
import FAQService from '../services/FAQService'

const FAQsList = props => {
    return (
        <div>
            <h3>Frequently Asked Questions:</h3>
            <table className="table">
                <thead>
                    <tr>
                        <th scope="col">Title</th>
                        <th scope="col">Question</th>
                        <th>&nbsp;</th>
                    </tr>
                    <tr>
                        <th> <input type="text" onChange={props.handleTitleChange} value={props.title} placeholder="Title" /> </th>
                        <th> <input type="text" onChange={props.handleQuestionChange} value={props.question} placeholder="Question" /> </th>
                        <th> <button type="button" onClick={props.updateFAQ} className="btn btn-primary btn-block">Save</button> </th>
                        <th> <button type="button" onClick={props.createFAQ} className="btn btn-primary btn-block">Create</button> </th>
                    </tr>
                </thead>
                <tbody>
                    {
                        props.faqs
                            .slice((props.page-1)*props.recordsNumber,
                                    Math.min(props.page*props.recordsNumber,
                                            props.faqs.length))
                            .filter(props.passesFilter)
                            .map(faq =>
                                <tr onClick={() => props.selectFAQ(faq.id)}  key={faq.id}>
                                    <td>{faq.title}</td>
                                    <td align="center">{faq.question}</td>
                                    <th> <button type="button" onClick={(e) => props.editFAQ(faq, e)} className="btn btn-primary btn-block">Edit</button>  </th>
                                    <th> <button type="button" onClick={(e) => props.deleteFAQ(faq, e)} className="btn btn-primary btn-block">Delete</button>    </th>
                                </tr>
                            )
                    }
                </tbody>
                <tfoot>
                    <tr>
                        <td>
                            <select onChange={props.handleRecordsNumberChange} value={props.recordsNumber}>
                                <option>10</option>
                                <option>25</option>
                                <option>50</option>
                                <option>100</option>
                                <option>All</option>
                            </select>
                        </td>
                        <td>
                            <ul className="pagination">
                                <li className="page-item"><a className="page-link" onClick={props.handlePageChange} disabled={props.page == 1}>Previous</a></li>
                                    {
                                        props.getPageNumbers().map(num =>
                                            <li key={num} className="page-item"><a className="page-link" onClick={props.handlePageChange}>{num}</a></li>
                                        )
                                    }
                                <li className="page-item"><a className="page-link" onClick={props.handlePageChange} disabled={props.page == Math.round(props.faqs.length/props.recordsNumber)}>Next</a></li>
                            </ul>
                        </td>
                        <td>
                            <button className="btn btn-secondary btn-block" onClick={props.handleFilterChange}>
                                {    
                                    (props.filter.question == undefined && props.filter.title == undefined) ? 'Search' : 'Clear Search'
                                }
                            </button>
                        </td>
                    </tr>
                </tfoot>
            </table>
        </div>
    )
}

export default FAQsList