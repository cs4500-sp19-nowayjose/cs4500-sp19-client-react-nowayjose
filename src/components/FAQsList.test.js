import React from 'react';
import renderer from 'react-test-renderer';
import FAQsList from './FAQsList';

test('renders faq correctly', function() {
  const component = renderer.create(
    <FAQsList
      faqs={[{"answers": [],"id": 2,"question": "Favorite color?","title": "Second question"}]}
      title=""
      question=""
      updateId=-1
      recordsNumber=10
      page=1
      filter={{}}

      editFAQ = {() => {}}
      deleteFAQ = {() => {}}
      createFAQ = {() => {}}
      updateFAQ = {() => {}}
      getPageNumbers = {() => {return [1]}}
      passesFilter = {() => {return true}}
      selectFAQ = {() => {}}
      handleTitleChange = {() => {}}
      handleQuestionChange = {() => {}}
      handlePageChange = {() => {}}
      handleRecordsNumberChange = {() => {}}
      handleFilterChange = {() => {}}
    />
  )
  const testInstance = component.root;
  expect(testInstance.findByProps({ className: "faq-row"})).toBeDefined();
  expect(testInstance.findByProps({ className: "faq-row"})).lengthToBe(1);
}) 