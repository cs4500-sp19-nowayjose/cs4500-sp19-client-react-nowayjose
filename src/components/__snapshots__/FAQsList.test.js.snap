// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Frequently Asked Questions:
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Question
        </th>
        <th>
          Â 
        </th>
      </tr>
      <tr>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
           
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="Question"
            type="text"
            value=""
          />
           
        </th>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
           
        </th>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="faq-row"
        onClick={[Function]}
      >
        <td>
          First question
        </td>
        <td
          align="center"
        >
          How old are you?
        </td>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
            
        </th>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
              
        </th>
      </tr>
      <tr
        className="faq-row"
        onClick={[Function]}
      >
        <td>
          Fourth question
        </td>
        <td
          align="center"
        >
          How many employees do you have?
        </td>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
            
        </th>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
              
        </th>
      </tr>
      <tr
        className="faq-row"
        onClick={[Function]}
      >
        <td>
          question
        </td>
        <td
          align="center"
        >
          Did you pass a background check?
        </td>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
            
        </th>
        <th>
           
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
              
        </th>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td>
          <select
            onChange={[Function]}
            value={10}
          >
            <option>
              10
            </option>
            <option>
              25
            </option>
            <option>
              50
            </option>
            <option>
              100
            </option>
            <option>
              All
            </option>
          </select>
        </td>
        <td>
          <ul
            className="pagination"
          >
            <li
              className="page-item"
            >
              <a
                className="page-link"
                disabled={true}
                onClick={[Function]}
              >
                Previous
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                onClick={[Function]}
              >
                1
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                disabled={false}
                onClick={[Function]}
              >
                Next
              </a>
            </li>
          </ul>
        </td>
        <td>
          <button
            className="btn btn-secondary btn-block"
            onClick={[Function]}
          >
            Search
          </button>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
